crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -462,31 +391,28 @@ where
         &self,
         mut acc: ChangedBalances,
         msg: &IbcMsgRecvPacket,
-        packet_data: &PacketData,
-        keys_changed: &BTreeSet<Key>,
+        ibc_trace: impl AsRef<str>,
+        amount: Amount,
     ) -> Result<ChangedBalances> {
         // If the transfer was a failure, then enable funds to
         // be withdrawn from the IBC internal address
-        if self.is_receiving_success(&msg.packet, keys_changed)? {
+        if self.is_receiving_success(
+            &msg.packet.port_id_on_a,
+            &msg.packet.chan_id_on_a,
+            msg.packet.seq_on_a,
+        )? {
             // Mirror how the IBC token is derived in
             // gen_ibc_shielded_transfer in the non-refund case
-            let ibc_denom = self.query_ibc_denom(
```

Comment:
Discussion here: https://github.com/anoma/namada/pull/3409#discussion_r1647526664 . Would the above comment about "mirror"ing still be correct with this change?

Reply:
I changed it to check the packet receipt